mesa/agent.py

```diff
@@ -188,15 +188,30 @@ def shuffle(self, inplace: bool = False) -> AgentSet:
 
         Returns:
             AgentSet: A shuffled AgentSet. Returns the current AgentSet if inplace is True.
+
+        Note:
+            Using inplace = True is more performant
+
         """
-        shuffled_agents = list(self)
-        self.random.shuffle(shuffled_agents)
+        weakrefs = list(self._agents.keyrefs())
+        self.random.shuffle(weakrefs)
+
+        if inplace:
+            self._agents.data = {entry: None for entry in weakrefs}
+            return self
+        else:
+            return AgentSet(
+                (agent for ref in weakrefs if (agent := ref()) is not None), self.model
```

Comment:
Cython doesn't support Walrus operator yet. Though not sure if AgentSet can/should be cythonized.

Reply:
Let's worry about that when we cross that bridge.

Reply:
Agreed, this now makes if faster. If no walrus + Cython make it even faster, we will do that when the time comes.